exchanges/retry/src/retryExchange.ts

```diff
@@ -0,0 +1,125 @@
+import {
+  makeSubject,
+  share,
+  pipe,
+  merge,
+  filter,
+  tap,
+  fromValue,
+  delay,
+  mergeMap,
+  takeUntil,
+} from 'wonka';
+import { Exchange, Operation, CombinedError } from '@urql/core';
+
+interface RetryExchangeOptions {
+  initialDelayMs?: number;
+  maxDelayMs?: number;
+  randomDelay?: boolean;
+  maxNumberAttempts?: number;
+  // Optional function to determine whether an error should be retried
+  retryIf?: (e: CombinedError) => boolean;
+}
+
+interface OperationWithRetry extends Operation {
+  retryCount?: number;
+}
+
+export const retryExchange = ({
+  initialDelayMs,
+  maxDelayMs,
+  randomDelay = true,
+  maxNumberAttempts,
+  retryIf,
+}: RetryExchangeOptions): Exchange => {
+  const MIN_DELAY = initialDelayMs || 1000;
+  const MAX_DELAY = maxDelayMs || 15000;
+  const MAX_ATTEMPTS = maxNumberAttempts || Infinity;
+
+  return ({ forward }) => ops$ => {
+    const sharedOps$ = pipe(ops$, share);
+    const { source: retry$, next: nextRetryOperation } = makeSubject<
+      OperationWithRetry
+    >();
+
+    const retryWithBackoff$ = pipe(
+      retry$,
+      mergeMap((op: OperationWithRetry) => {
```

Comment:
I believe is better avoid name abbreviations.

Reply:
We use the `op` for "operation" abbreviation in multiple exchanges in the codebase already. The type annotation makes the intent clear in most cases anyway :)